src/lib/actions/helpers/decorators/__tests__/decorateHtmlCode.test.js

```diff
@@ -0,0 +1,195 @@
+    );
+  });
+
+  it('returns the decorated code with replaced callback ids if they exist', function() {
+    var settings = {
+      language: 'html',
+      source:
+        '<script>_satellite._onCustomCodeSuccess(${reactorCallbackId})</script>'
+    };
+
+    var decorateHtmlCode = decorateHtmlCodeInjector();
+    var decoratedResult = decorateHtmlCode(
+      {
+        settings: settings,
+        event: {}
+      },
+      settings.source
+    );
+
+    expect(decoratedResult.code).toBe(
+      '<script>_satellite._onCustomCodeSuccess(0)</script>'
+    );
+  });
+
+  it('does not replace data element tokens for an embedded html action', function() {
+    var settings = {
+      language: 'html',
+      source: '<div>%productname%</div>'
+    };
+
+    var decorateHtmlCode = decorateHtmlCodeInjector();
+    var decoratedResult = decorateHtmlCode(
+      {
+        settings: settings,
+        event: {},
+        relatedElement: {}
+      },
+      settings.source
+    );
+
+    expect(decoratedResult.code).toBe('<div>%productname%</div>');
+    expect(mockTurbine.replaceTokens).not.toHaveBeenCalled();
+  });
+
+  it('does replace data element tokens for an html action loaded from a file', function() {
+    var settings = {
+      language: 'html',
+      source: 'url1',
+      isExternal: true
+    };
+
+    var decorateHtmlCode = decorateHtmlCodeInjector();
+    var decoratedResult = decorateHtmlCode(
+      {
+        settings: settings,
+        event: {},
+        relatedElement: {}
+      },
+      '<div>%productname%</div>'
+    );
+
+    expect(decoratedResult.code).toBe('<div>replaced - productname</div>');
+  });
+
+  it(
+    'returns a resolved promise on the promise key when HTML code ' +
+      'does not contain callbacks',
+    function() {
+      var settings = {
+        language: 'html',
+        source: '<script>console.log("logging")</script>'
+      };
+
+      var p = Promise.resolve();
+      var decorateHtmlCode = decorateHtmlCodeInjector({
+        '@adobe/reactor-promise': {
+          resolve: function() {
+            return p;
+          }
+        }
+      });
+
+      var decoratedResult = decorateHtmlCode(
+        {
+          settings: settings
+        },
+        settings.source
+      );
+
+      expect(decoratedResult.promise).toBe(p);
+    }
+  );
+
+  it(
+    'returns a promise that will be resolved when HTML code contains callbacks' +
+      'and _satellite._onCustomCodeSuccess is called',
+    function(done) {
+      var settings = {
+        language: 'html',
+        source:
+          '<script>_satellite._onCustomCodeSuccess(${reactorCallbackId})</script>'
+      };
+
+      var decorateHtmlCode = decorateHtmlCodeInjector();
+
+      decorateHtmlCode(
+        {
+          settings: settings
+        },
+        settings.source
+      ).promise.then(function() {
+        done();
+      });
+
+      window._satellite._onCustomCodeSuccess(0);
+    }
```

Comment:
This test should probably be improved. As the test is, in isolation, we don't know if the promise is being resolved _as a result of_ `window._satellite.onCustomCodeSuccess(0)` being called or if it was resolved already. Here's how we have approached this sort of problem in Alloy: We have a [`flushPromiseChains` helper](https://github.com/adobe/alloy/blob/master/test/unit/helpers/flushPromiseChains.js). Then we would form our test like this:

```
var onPromiseResolved = jasmine.createSpy("onPromiseResolved");
var decorateCodePromise = decorateHtmlCode({
  settings: settings
}, settings.source).promise;

decorateCodePromise.then(onPromiseResolve);

return flushPromiseChains().then(function() {
  expect(onPromiseResolved).not.toHaveBeenCalled();
  window._satellite._onCustomCodeSuccess(0);
  return decorateCodePromise;
})
```